Added custom sampler support based on action in request (#10136)
opensearch-project/OpenSearch: CHANGELOG.md

- Restore support for Java 8 for RestClient ([#11562](https://github.com/opensearch-project/OpenSearch/pull/11562))
- Add deleted doc count in _cat/shards ([#11678](https://github.com/opensearch-project/OpenSearch/pull/11678))
- Capture information for additional query types and aggregation types ([#11582](https://github.com/opensearch-project/OpenSearch/pull/11582))

> - Add override support for sampling based on action ([#9621](https://github.com/opensearch-project/OpenSearch/issues/9621))

Reviewer: ```suggestion
- Added custom sampler support based on action in reques ([#9621](https://github.com/opensearch-project/OpenSearch/issues/9621))
```
